web/src/settings_users.ts

```diff
@@ -191,29 +191,49 @@ function role_selected_handler(
     widget.render();
 }
 
-function get_roles(): {
+function count_users_by_role(user_ids: number[]): Record<number, number> {
+    const role_counts: Record<number, number> = {};
+
+    for (const user_id of user_ids) {
+        const user = people.get_by_user_id(user_id);
+        const role_code = user.role;
+
+        role_counts[role_code] = (role_counts[role_code] ?? 0) + 1;
+    }
+
+    return role_counts;
+}
+
+type RoleOption = {
     unique_id: number;
     name: string;
-}[] {
+};
+
+function get_roles_with_counts(user_ids: number[]): RoleOption[] {
+    const role_counts = count_users_by_role(user_ids);
     return [
-        {unique_id: 0, name: $t({defaultMessage: "All roles"})},
+        {
+            unique_id: 0,
+            name: `${$t({defaultMessage: "All roles"})} (${user_ids.length})`, // Total count for all users in this list
+        },
         ...Object.values(settings_config.user_role_values)
             .map((user_role_value) => ({
                 unique_id: user_role_value.code,
-                name: user_role_value.description,
+                name: `${user_role_value.description} (${role_counts[user_role_value.code] ?? 0})`, // Role-specific count
```

Comment:
You can remove the inline comments; I think what they are saying is obvious. BTW our style is to not write inline comments; they should always be on the line before what they document.